app/models/host_validator.rb

```diff
@@ -26,7 +32,24 @@ def invalid_tld?
     INVALID_TLDS.include?(File.extname(host))
   end
 
+  def invalid_ip?
+    if is_ip?(host)
+      INVALID_IPS.any? { |invalid_ip| invalid_ip.include?(host) }
+    else
+      false
+    end
+  end
+
   def host
     URI.parse(url).host
   end
+
+  def is_ip?(host)
+    begin
+      IPAddr.new(host)
+      true
+    rescue IPAddr::InvalidAddressError
```

Comment:
`IPAddr.new` throws this error if it's an invalid IP. We could try matching by using internal regex constants like `IPAddr::RE_IPV4ADDRLIKE` / `IPAddr::RE_IPV6ADDRLIKE_COMPRESSED` / `IPAddr::RE_IPV6ADDRLIKE_FULL` but I'm OK with this as-is.